go/tools/bzltestutil/lcov.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2022 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bzltestutil
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// ConvertCoverToLcov converts the go coverprofile file coverage.dat.cover to
+// the lcov format and stores it in coverage.dat, where it is picked up by
+// Bazel.
+// The conversion emits line and branch coverage, but not function coverage.
+func ConvertCoverToLcov() error {
+	inPath := flag.Lookup("test.coverprofile").Value.String()
+	outPath := strings.TrimSuffix(inPath, ".cover")
```

Comment:
I will address the other comments, but we I don't think we have any flexibility in the `outPath`: `inPath` is simply Bazel's `COVERAGE_OUTPUT_FILE` with an additional `.cover` suffix and the output report has to be placed at `COVERAGE_OUTPUT_FILE` for Bazel to pick it up. Currently, `go_test` uses a fake `lcov_merger`, meaning that Bazel will only see the report if it has exactly that name rather than merging all `*.dat` files in its directory.